crates/ironrdp-connector/src/connection.rs

```diff
@@ -782,24 +782,22 @@ impl Sequence for ClientConnector {
 fn create_gcc_blocks(config: &Config, selected_protocol: nego::SecurityProtocol) -> gcc::ClientGccBlocks {
     use ironrdp_pdu::gcc::*;
 
-    let color_depth = config
-        .bitmap
-        .as_ref()
-        .map(|bitmap| match bitmap.color_depth {
-            15 => SupportedColorDepths::BPP15,
-            16 => SupportedColorDepths::BPP16,
-            24 => SupportedColorDepths::BPP24,
-            32 => SupportedColorDepths::BPP32,
-            _ => panic!("Unsupported color depth: {}", bitmap.color_depth),
-        })
-        .unwrap_or(SupportedColorDepths::BPP16);
+    let max_color_depth = config.bitmap.as_ref().map(|bitmap| bitmap.color_depth).unwrap_or(32);
+
+    let supported_color_depths = match max_color_depth {
+        15 => SupportedColorDepths::BPP15,
+        16 => SupportedColorDepths::BPP16,
+        24 => SupportedColorDepths::BPP24,
+        32 => SupportedColorDepths::BPP32 | SupportedColorDepths::BPP16,
+        _ => panic!("Unsupported color depth: {}", max_color_depth),
+    };
 
     ClientGccBlocks {
         core: ClientCoreData {
             version: RdpVersion::V5_PLUS,
             desktop_width: config.desktop_size.width,
             desktop_height: config.desktop_size.height,
-            color_depth: ColorDepth::Bpp4, // ignored because we use the optional core data below
+            color_depth: ColorDepth::Bpp8, // ignored because we use the optional core data below
```

Comment:
Curiosity: is there any meaning to this change since it’s ignored?

Reply:
it's defined as ignored by the spec, but even in those cases it's better to stick to the same values sent by the original client, just in case (that's what FreeRDP and mstsc send). Even if they are truly ignored, it helps removing noise when trying to compare packets for differences that should matter.